src/main/java/build/buildfarm/server/ActionCacheService.java

```diff
@@ -38,7 +38,7 @@ public void getActionResult(
     try {
       instance = instances.getInstance(request.getInstanceName());
     } catch (InstanceNotFoundException ex) {
-      responseObserver.onError(new StatusException(Status.NOT_FOUND));
```

Comment:
My personal preference is to do something like this:
responseObserver.onError(BuildFarmInstances.exceptionToError(ex))

Passing in the full observer means that the method could do all sorts of weird shenanigans, and is also more difficult to test.

Reply:
Done.